zerver/tests/test_message_notification_emails.py

```diff
@@ -1193,6 +1208,10 @@ def test_realm_message_content_allowed_in_email_notifications(self) -> None:
         do_change_user_setting(
             user, "message_content_in_email_notifications", False, acting_user=None
         )
+        # Need to clear cache here after changing the setting because code
+        # to clear the cache is not executed as it is called in on_commit
+        # callback.
+        delete_user_profile_caches([user], user.realm_id)
```

Comment:
This is needed now to make sure cache is cleared as later on `handle_missedmessage_emails` calls `get_user_profile_by_id` which uses the cached value of user object. Previously we called `save()` and `flush_user_profile` was called but now we use `bulk_update` and cache is cleared only using `on_commit` callback which is not called in tests.